core/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala

```diff
@@ -709,6 +711,17 @@ trait CDCReaderImpl extends DeltaLogging {
     ChangeDataFeedTableFeature.metadataRequiresFeatureToBeEnabled(metadata, spark)
   }
 
+  def assertNoDeletionVectors(version: Long, file: FileAction): Unit = {
```

Comment:
Rather than checking if the file has DV or not, why not just fail in the beginning if the table has DV enabled?

Reply:
That will block more tables but make the check more reliable. I'll refactor it.